### PR TITLE
my_inject works - as long as you don't do silly stuff

### PR DESCRIPTION
**Pull request by TStrothjohann**
The symbol-variant isn't quite done.

### CODE REVIEW

File: spec/array_spec.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
+require 'array'
+
+describe Array do
+
+    array = [1,2,3,4]
```
Comment (spike01) [inline spec/array_spec.rb:5]: This is best enclosed in a `let`

File: lib/array.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  end
+
+  def get_proc_inside(&proc)
+    proc.call(2, 2)
+  end
+
+  def my_inject(memo=nil, &proc)
+    array = self
+    memo ||=array.shift
+    array.size.times {memo = proc.call(memo,array.shift)}
```
Comment (spike01) [inline lib/array.rb:30]: `array.size.times` => `array.each`

File: spec/array_spec.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  it "can sum up all values without map" do
+    array = [2,2,3,4]
+    expect(array.my_sum_up).to eq(11)
+    
+  end
+
+  it "can use the code-block I pass in" do
+    expect(array.get_proc_inside{|memo, item| memo + item}).to eq(4)
+  end
+
```
Comment (spike01) [inline spec/array_spec.rb:27]: These tests don't test your actual inject method!

Comment (TStrothjohann) [inline spec/array_spec.rb:27]: Oh. Sorry. I did not delete my brainstorming.

File: lib/array.rb (hunk 2 above)
Comment (spike01) [inline lib/array.rb:30]: Also think about `yield`

File: spec/array_spec.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  it "can add each of the negative value out of the array to memo" do
+    expect([1, -2, 3, -4].my_inject{|memo, item| memo + item}).to eq -2
+    expect([1, -2, 3, -4].inject{|memo, item| memo + item}).to eq -2
+  end
+
+  it "can multiply each of the array to memo" do
+    expect([1, 2, 3, 4].my_inject{|memo, item| memo * item}).to eq 24
+    expect([1, 2, 3, 4].inject{|memo, item| memo * item}).to eq 24
+  end
+
```
Comment (spike01) [inline spec/array_spec.rb:43]: Test directly against the real inject method

File: spec/array_spec.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    
+  end
+
+  it "can use the code-block I pass in" do
+    expect(array.get_proc_inside{|memo, item| memo + item}).to eq(4)
+  end
+  
+
+  it "can add each of the array to memo" do
+    expect([1, 2, 3, 4].my_inject{|memo, item| memo + item}).to eq 10
```
Comment (spike01) [inline spec/array_spec.rb:30]: Declare this array as a `let`

**Comment by spike01** [comment]
Good inject method. Remember to clean up your code as you go along - dead code should be deleted!

**Pull request closed without merging**